src/sentry/integrations/services/integration/impl.py

```diff
@@ -357,6 +358,56 @@ def update_organization_integration(
         )
         return ois[0] if len(ois) > 0 else None
 
+    def start_grace_period_for_provider(
+        self,
+        *,
+        organization_id: int,
+        provider: str,
+        grace_period_end: datetime,
+        status: int = ObjectStatus.ACTIVE,
+        skip_oldest: bool = False,
+    ) -> list[RpcOrganizationIntegration]:
+        filter_kwargs = {
+            "organization_id": organization_id,
+            "integration__provider": provider,
+        }
+        if status is not None:
+            filter_kwargs["status"] = status
```

Comment:
status is never `None` now? I think you can put the old code back if you always want to pass a status (put filter kwargs directly into `OrganizationIntegration.objects.filter`